android-stub/src/main/java/com/android/org/conscrypt/OpenSSLSocketImpl.java

```diff
@@ -13,266 +13,115 @@
-import javax.net.ssl.HandshakeCompletedListener;
 import javax.net.ssl.SSLException;
 import javax.net.ssl.SSLSession;
 
-/**
- * Implementation of the class OpenSSLSocketImpl based on OpenSSL.
- * <p>
- * Extensions to SSLSocket include:
- * <ul>
- * <li>handshake timeout
- * <li>session tickets
- * <li>Server Name Indication
- * </ul>
- */
-public class OpenSSLSocketImpl
-        extends javax.net.ssl.SSLSocket implements NativeCrypto.SSLHandshakeCallbacks {
-    protected OpenSSLSocketImpl(SSLParametersImpl sslParameters) throws IOException {
-        throw new RuntimeException("Stub!");
-    }
-
-    protected OpenSSLSocketImpl(SSLParametersImpl sslParameters, String[] enabledProtocols,
-            String[] enabledCipherSuites) throws IOException {
-        throw new RuntimeException("Stub!");
-    }
-
-    protected OpenSSLSocketImpl(String host, int port, SSLParametersImpl sslParameters)
-            throws IOException {
-        throw new RuntimeException("Stub!");
-    }
-
-    protected OpenSSLSocketImpl(InetAddress address, int port, SSLParametersImpl sslParameters)
-            throws IOException {
-        throw new RuntimeException("Stub!");
-    }
-
-    protected OpenSSLSocketImpl(String host, int port, InetAddress clientAddress, int clientPort,
-            SSLParametersImpl sslParameters) throws IOException {
-        throw new RuntimeException("Stub!");
-    }
-
-    protected OpenSSLSocketImpl(InetAddress address, int port, InetAddress clientAddress,
-            int clientPort, SSLParametersImpl sslParameters) throws IOException {
-        throw new RuntimeException("Stub!");
-    }
-
-    protected OpenSSLSocketImpl(Socket socket, String host, int port, boolean autoClose,
-            SSLParametersImpl sslParameters) throws IOException {
-        throw new RuntimeException("Stub!");
-    }
-
-    @Override
-    public synchronized void startHandshake() throws IOException {
-        throw new RuntimeException("Stub!");
-    }
-
-    @SuppressWarnings("unused") // used by NativeCrypto.SSLHandshakeCallbacks / client_cert_cb
-    @Override
-    public void clientCertificateRequested(byte[] keyTypeBytes, byte[][] asn1DerEncodedPrincipals)
-            throws CertificateEncodingException, SSLException {
-        throw new RuntimeException("Stub!");
-    }
-
-    @SuppressWarnings("unused") // used by NativeCrypto.SSLHandshakeCallbacks / info_callback
-    @Override
-    public void handshakeCompleted() {
-        throw new RuntimeException("Stub!");
-    }
-
-    @SuppressWarnings("unused") // used by NativeCrypto.SSLHandshakeCallbacks
-    @Override
-    public void verifyCertificateChain(byte[][] bytes, String authMethod)
-            throws CertificateException {
-        throw new RuntimeException("Stub!");
-    }
-
-    @Override
-    public InputStream getInputStream() throws IOException {
-        throw new RuntimeException("Stub!");
-    }
+public abstract class OpenSSLSocketImpl extends AbstractConscryptSocket {
```

Comment:
Oh, this isn't going to work. You're changing the class hierarchy, but this class has already been shipped on Android 4.4 KitKat. This is supposed to act as a stub for what already exists, so you can't redefine where the methods exist or the class hierarchy.